lib/vector-core/src/event/metric/mod.rs

```diff
@@ -0,0 +1,1008 @@
+            .is_some()
+    }
+
+    /// Returns the string value of a tag, if it exists
+    pub fn tag_value(&self, name: &str) -> Option<String> {
+        self.tags().and_then(|t| t.get(name).cloned())
+    }
+
+    /// Inserts a tag into this metric.
+    ///
+    /// If the metric did not have this tag, `None` will be returned. Otherwise, `Some(String)` will be returned,
+    /// containing the previous value of the tag.
+    ///
+    /// *Note:* This will create the tags map if it is not present.
+    pub fn insert_tag(&mut self, name: String, value: String) -> Option<String> {
+        self.series.insert_tag(name, value)
+    }
+
+    /// Gets the given tag's corresponding entry in this metric.
+    ///
+    /// *Note:* This will create the tags map if it is not present, even if nothing is later inserted.
+    pub fn tag_entry(&mut self, key: String) -> btree_map::Entry<String, String> {
+        self.series.tag_entry(key)
+    }
+
+    /// Zeroes out the data in this metric.
+    pub fn zero(&mut self) {
+        self.data.zero();
+    }
+
+    /// Adds the data from the `other` metric to this one.
+    ///
+    /// The other metric must be incremental and contain the same value type as this one.
+    #[must_use]
+    pub fn add(&mut self, other: impl AsRef<MetricData>) -> bool {
+        self.data.add(other.as_ref())
+    }
+
+    /// Updates this metric by adding the data from `other`.
+    #[must_use]
+    pub fn update(&mut self, other: impl AsRef<MetricData>) -> bool {
+        self.data.update(other.as_ref())
+    }
+
+    /// Subtracts the data from the `other` metric from this one.
+    ///
+    /// The other metric must contain the same value type as this one.
+    #[must_use]
+    pub fn subtract(&mut self, other: impl AsRef<MetricData>) -> bool {
+        self.data.subtract(other.as_ref())
+    }
+}
+
+impl AsRef<MetricData> for Metric {
+    fn as_ref(&self) -> &MetricData {
+        &self.data
+    }
+}
+
+impl AsRef<MetricValue> for Metric {
+    fn as_ref(&self) -> &MetricValue {
+        &self.data.value
+    }
+}
+
+impl Display for Metric {
+    /// Display a metric using something like Prometheus' text format:
+    ///
+    /// ```text
+    /// TIMESTAMP NAMESPACE_NAME{TAGS} KIND DATA
+    /// ```
+    ///
+    /// TIMESTAMP is in ISO 8601 format with UTC time zone.
+    ///
+    /// KIND is either `=` for absolute metrics, or `+` for incremental
+    /// metrics.
+    ///
+    /// DATA is dependent on the type of metric, and is a simplified
+    /// representation of the data contents. In particular,
+    /// distributions, histograms, and summaries are represented as a
+    /// list of `X@Y` words, where `X` is the rate, count, or quantile,
+    /// and `Y` is the value or bucket.
+    ///
+    /// example:
+    /// ```text
+    /// 2020-08-12T20:23:37.248661343Z vector_processed_bytes_total{component_kind="sink",component_type="blackhole"} = 6391
+    /// ```
```

Comment:
Nice comment. Tell you what, this is a lot better than computing in one's head what the display does.

Reply:
All credit goes to whoever the heck wrote it first.  This just looks like I did because of all the rearranging the PR is doing.

Reply:
Did some digging, and not surprisingly, this great code comment (and its sibling comments on the other `Display` impls for related types) is courtesy of @bruceg. :)